### PR TITLE
Enhance/search by type

### PR DESCRIPTION
**Pull request by NickEngmann**
Way to search by type and display deck collections page accordingly.
Fetch all collections on ComponentWillMount and build options for Typeahead.
If user selects a Deck collection and either hits enter or selects the search icon they are redirected via react-router to new deck collections page

Mobile:
image
Desktop:
image

image

### CODE REVIEW

File: src/app/decks/Decks.js [hunk 1]
```diff
@@ -75,10 +86,28 @@ class Decks extends Component {
...
+    }
+  }
+
   render() {
     const { location } = this.props;
     const { decks, isLoading, isError, savedDecks, activeTab } = this.state;
 
+    if (this.state.redirect) {
+      for(var i = 0; i< this.state.collections.length; i++){
+        if(this.state.collections[i].name === this.state.searchString){
```
Comment (NickEngmann) [inline src/app/decks/Decks.js]: I'm sure there is a better way to do this besides iterating over the array to find the associated id, but I couldn't find one. Any help here @nlaz ?

File: src/app/decks/Decks.js [hunk 2]
```diff
@@ -123,6 +156,22 @@ class Decks extends Component {
...
+              options={filteredNames}
+              onChange={(e) => {
+                this.setState({searchString: e[0]});
+              }}
+              placeholder="Search..."
+            />
+            <button type="submit" className="search-button text-white bg-primary border-info" onClick={this.handleSearch}>
+                <i className="fa fa-search"></i>
+            </button>
+        </div>
```
Comment (nlaz) [inline src/app/decks/Decks.js]: Can you make the Searchbar into its own component? Would help with reuse in the future.

**Comment by nlaz** [comment]
@NickEngmann The ideal experience of the search bar is to surface topics that are relative to the searched term right? Simply returning a list of categories that fit that string isn't very helpful at the moment.

Instead of fetching all categories upfront and filtering them, lets either 1) filter all decks on the current page based on deck name and description, or 2) create a search endpoint and return mixed results after a user starts searching. Support for fuzzy searching would add even more value to this feature. We can start with #1 and move to #2 as we build our backend. What do you think?

File: src/app/decks/Decks.js (hunk 1 above)
Comment (NickEngmann) [reply src/app/decks/Decks.js]: Thanks! 👍 👍 👍

**Comment by nlaz** [comment]
To help out, I moved the search bar to the header and stripped out the styling for now (because I'm boring).
screen shot 2018-09-05 at 11 28 51 am

**Comment by nlaz** [comment]
I probably should've shared what I was thinking for finding the deck collections pages. Right now, there is no way to discover them. That's okay. Simply sharing the link will work for us for now. We will add a section at the bottom of the homepage to find collections. Hopefully we can do that soon.

For the searchbar, it should find the most relevant decks instead of collections. I'm okay with backlogging this for now as it will take us some time to get it right. Closing this PR until we are ready again.

**Pull request closed without merging**